### ISSUE TITLE
Background fetch not working

### ISSUE THREAD

**Issue by phiren** [paulw11/BackgroundFetchTest#1]
I have installed your code in iPhone 7 using debug profile, but it is not giving any notification alert, seems it not working at all,

Background refresh is enabled on my device.

**Comment by paulw11** [comment]
How long did you wait?  You will probably have to wait several hours before the first fetch will occur.  

Have you tried Debug->Simulate background fetch in Xcode? - if you do this when running on a real device then an exception is triggered in the debugger, but pressing the |> continue button should result in the notification being displayed.

**Comment by phiren** [comment]
Yes, i have tried with simulate background fetch in Xcode, it almost 30 hours before i installed the app

**Comment by paulw11** [comment]
Have you opened the app from your device (i.e. not from Xcode) a couple of times?  It is working for me.

**Comment by phiren** [comment]
I have reset my device settings, but not sure whats going wrong. It not working for any app.

**Comment by phiren** [comment]
Simulating background fetch from Xcode always works

**Comment by paulw11** [comment]
Open the app from your device, not Xcode and then wait approximately 48 hours; you should get regular background fetch events.  The app straight from the repo is working for me on iOS 10.3

**Comment by phiren** [comment]
I have installed on other device, it works fine, but somehow not working on my device, none of the app with background fetch works.
